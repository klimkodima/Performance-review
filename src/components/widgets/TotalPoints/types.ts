export type TTotalPoints = {
  totalPoints: number;
  level: number;
  premium: number;
};
