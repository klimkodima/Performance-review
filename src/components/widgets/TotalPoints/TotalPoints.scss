.total-points-container {
  width: 100%;
  height: 95%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 40px 60px;
  row-gap: 16px;
  margin: 5% 0 !important;
  background-color: #f6f6f6;
  border: 2px #d7d7d7 solid;
  border-radius: 5px;
  box-sizing: border-box !important;
}
.total-points-item-wrapper{
  display: flex;
  width: 80%;
  flex-direction: column;
  align-items: left;
}

.total-points-item {
  width: 35%;
  height: 100%;
  display: flex;
  position: relative;
  flex-direction: column;
  align-items: center;
  padding-top: 25px;
}

.total-points-label {
  font-size: 1.5em;
  font-weight: bold;
  color: #7e7e7e;
}

.total-points-value {
  display: flex;
  width: 100%;
  height: 120px;
  background-color: white;
  border-radius: 10px;
  border: 2px #d7d7d7 solid;
  font-size: 26px;
  align-items: center;
  justify-content: center;
  font-weight: bold;
  color: #f54470;
  margin-top: 20px;
}
