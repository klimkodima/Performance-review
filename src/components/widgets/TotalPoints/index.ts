export { default as TotalPoints } from './TotalPointsContainer';
