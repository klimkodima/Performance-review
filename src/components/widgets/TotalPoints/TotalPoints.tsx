import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import Grid from '@mui/material/Grid';

import { TTotalPoints } from './types';

import './TotalPoints.scss';

const TotalPoints: FC = () => {
  const { t } = useTranslation('common', { keyPrefix: 'totalPoints' });
  const data: TTotalPoints = {
    totalPoints: 826,
    level: 2,
    premium: 10
  };
  const formatLevel = (level: number): string => {
    switch (level) {
      case 1:
        return `${level}st`;
      case 2:
        return `${level}nd`;
      case 3:
        return `${level}rd`;
      default:
        return `${level}th`;
    }
  };

  return (
    <Grid item xs={4} data-testid='total-container'>
      <div className='total-points-container'>
        <div className='total-points-item-wrapper'>
          <span className='total-points-label'>{t('Total Points')}</span>
          <span className='total-points-value'>{data.totalPoints}</span>
        </div>
        <div className='total-points-item-wrapper'>
          <span className='total-points-label'>{t('Level position')}</span>
          <span className='total-points-value'>{formatLevel(data.level)}</span>
        </div>
        <div className='total-points-item-wrapper'>
          <span className='total-points-label'>{t('Seniority Premium')}</span>
          <span className='total-points-value'>
            {data.premium === 0 ? '-' : data.premium}%
          </span>
        </div>
      </div>
    </Grid>
  );
};

export default TotalPoints;
