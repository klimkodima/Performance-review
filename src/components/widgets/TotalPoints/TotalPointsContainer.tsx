import { FC } from 'react';
import TotalPoints from './TotalPoints';
import { TTotalPoints } from './types';

const TotalPointsContainer: FC = () => {
  const data: TTotalPoints = {
    totalPoints: 826,
    level: 2,
    premium: 10
  };

  return <TotalPoints data={data} />;
};

export default TotalPointsContainer;
