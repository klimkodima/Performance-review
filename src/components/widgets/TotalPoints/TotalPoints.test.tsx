import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import TotalPoints from './TotalPoints';

describe('TotalPoints widget container tests', () => {
  it('TotalPoints widget snapshot', () => {
    const component = renderer.create(<TotalPoints />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('TotalPoints widgets component rendered', () => {
    render(<TotalPoints />);
    const container = screen.getByTestId('total-container');
    expect(container).toBeInTheDocument();
  });
  it('TotalPoints widgets component has 3 items', () => {
    const { container } = render(<TotalPoints />);
    expect(
      container.getElementsByClassName('total-points-item-wrapper')
    ).toHaveLength(3);
  });
});

describe('TotalPoints widget item tests', () => {
  xit('TotalPoints widgets item rendered', () => {
    render(<TotalPoints />);
    const item = screen.getAllByTestId('criteria-item');
    expect(item[0]).toBeInTheDocument();
  });
  xit('TotalPoints widgets item has classes', () => {
    render(<TotalPoints />);
    const item = screen.getAllByTestId('criteria-item');
    expect(item[1]).toHaveClass('criteria-item');
  });
});

describe('Criteria widget item title tests', () => {
  xit('Criteria widgets item title rendered', () => {
    render(<TotalPoints />);
    const itemTitle = screen.getAllByTestId('criteria-item-title');
    expect(itemTitle[1]).toBeInTheDocument();
  });
  xit('Criteria widgets item title has classes', () => {
    render(<TotalPoints />);
    const itemTitle = screen.getAllByTestId('criteria-item-title');
    expect(itemTitle[1]).toHaveClass('criteria-item-title');
  });
  xit('Criteria widgets item title has text', () => {
    render(<TotalPoints />);
    const itemTitle = screen.getAllByTestId('criteria-item-title');
    expect(itemTitle[1]).toContainHTML('Support/Audits Hours');
  });
});

describe('Criteria widget item statistics tests', () => {
  xit('Criteria widgets item statistic rendered', () => {
    render(<TotalPoints />);
    const itemStatistics = screen.getAllByTestId('criteria-item-statistic');
    expect(itemStatistics[1]).toBeInTheDocument();
  });
  xit('Criteria widgets item statistics has classes', () => {
    render(<TotalPoints />);
    const itemStatistics = screen.getAllByTestId('criteria-item-statistic');
    expect(itemStatistics[1]).toHaveClass('criteria-item-statistic');
  });
  xit('Criteria widgets item statistics has text', () => {
    render(<TotalPoints />);
    const itemStatistics = screen.getAllByTestId('criteria-item-statistic');
    expect(itemStatistics[1]).toContainHTML('162');
  });
});
