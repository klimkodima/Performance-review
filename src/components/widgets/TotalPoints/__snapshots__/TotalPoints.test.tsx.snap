// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalPoints widget container tests TotalPoints widget snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-13i4rnv-MuiGrid-root"
  data-testid="total-container"
>
  <div
    className="total-points-container"
  >
    <div
      className="total-points-item-wrapper"
    >
      <span
        className="total-points-label"
      >
        Total Points
      </span>
      <span
        className="total-points-value"
      >
        826
      </span>
    </div>
    <div
      className="total-points-item-wrapper"
    >
      <span
        className="total-points-label"
      >
        Level position
      </span>
      <span
        className="total-points-value"
      >
        2nd
      </span>
    </div>
    <div
      className="total-points-item-wrapper"
    >
      <span
        className="total-points-label"
      >
        Seniority Premium
      </span>
      <span
        className="total-points-value"
      >
        10
        %
      </span>
    </div>
  </div>
</div>
`;
